### PR TITLE
Add expand/collapse grouping events

### LINKED ISSUE #462: Data View events for group expand/collapse?
The current version of the data view has no event hooks for when groupings are expanded/collapsed. 
I have code already written for this if you'd accept a pull request?

### PR DESCRIPTION
**Pull request by awood20**
Updated the dataview.js to add Slick.Event for expand and collapse actions within data view groupings. fixes #462

### CODE REVIEW

**Comment by ghiscoding** [comment]
@awood20 
Could you please fix your indentation, it should be 2 spaces, that is our default, so that we see less differences in the PR

**Comment by ghiscoding** [comment]
@awood20 
Sorry to say this but we still see a lot of differences in the PR, you indentation is misaligned in a few area and that shows as differences in the PR. Which editor are you using? I personally use [Visual Studio Code](https://code.visualstudio.com/) and you could set the auto-format and that would fix all your indentation. 

Also note, you still have some `let` instead of `var`, it would be better to stick with `var`

If it's too much work for you, just let me know and I can fix your PR in coming days.

**Comment by awood20** [comment]
Hi, sorry. I'll have a look at the changes today again. It may have been visual studio that has messed the file.

**Comment by awood20** [comment]
I have taken the master branch code and re-applied the changes outside of Visual Studio. If the spacing is messed now, I'm sure what's going on. Removed out the remaining let as well.

File: slick.dataview.js [hunk 1]
```diff
@@ -542,36 +556,40 @@
       refresh();
     }
 
-    /**
-     * @param varArgs Either a Slick.Group's "groupingKey" property, or a
-     *     variable argument list of grouping values denoting a unique path to the row.  For
-     *     example, calling collapseGroup('high', '10%') will collapse the '10%' subgroup of
-     *     the 'high' group.
-     */
-    function collapseGroup(varArgs) {
```
Comment (ghiscoding) [inline slick.dataview.js]: why did you remove the comments and the `varArgs`? If you thought this is not used at all, it's not completely true, it's used below with the `arguments` but doesn't call the `varArgs` directly, however the comments is still valid. So to remove confusion, the create added the `varArgs` for users who call this method, so I think it should all go back to it's original state, which is comments + arg.

File: slick.dataview.js [hunk 2]
```diff
@@ -542,36 +556,40 @@
...
+      onGroupCollapsed.notify({level:level, groupingKey:groupingKey});
     }
 
-    /**
-     * @param varArgs Either a Slick.Group's "groupingKey" property, or a
-     *     variable argument list of grouping values denoting a unique path to the row.  For
-     *     example, calling expandGroup('high', '10%') will expand the '10%' subgroup of
-     *     the 'high' group.
-     */
-    function expandGroup(varArgs) {
```
Comment (ghiscoding) [inline slick.dataview.js]: same as what I wrote earlier, plz don't remove the comment and args, they serve a purpose and are valid.

File: slick.dataview.js (hunk 1 above)
Comment (awood20) [reply slick.dataview.js]: If you wish to put the comments back in I can. The removal of the parameter was because it wasn't being used. So really I was just tidying code with those removals.

File: slick.dataview.js (hunk 2 above)
Comment (awood20) [reply slick.dataview.js]: Having unused parameters is confusing tbh. Looks like it's just been left or missed, rather than deliberate.

Comment (ghiscoding) [reply slick.dataview.js]: not if you call the method and your editor shows you the `varArgs` and the comment with intellisense, then you'll be happy to have it

Comment (awood20) [reply slick.dataview.js]: In that case, should you not be using the named parameter within the function?

Comment (ghiscoding) [reply slick.dataview.js]: you can't, it's using `arguments` because it can be 1 or more and `arguments` is the standard way to call it. It just sux for commenting purposes though and that is why we have this conversation.

Comment (awood20) [reply slick.dataview.js]: OK I'll add it back in. Will be later today though before I can get back to it.

File: slick.dataview.js [hunk 3]
```diff
@@ -511,14 +513,26 @@
     }
 
     function expandCollapseAllGroups(level, collapse) {
-      if (level == null) {
+      if (level === null) {
```
Comment (ghiscoding) [inline slick.dataview.js]: hmm I see you've put all equality with triple equal but I wonder if this one in particular will break something. If the `level` can be `undefined`, then this will be broken

**Comment by ghiscoding** [comment]
@awood20 
I'm ok with most of your PR except that change where you added the triple equality check, I'm afraid this might break some code if the value is `undefined` it won't be covered, I don't know if that can ever happen but that could be a potential break compare to previous code. Could you fix it before we can merge? Thanks

**Comment by 6pac** [comment]
Thanks for working through it all, both of you.

@awood20 I've got a lint going as well that constantly complains about ==, but in many places we are deliberately using lazy (==) comparison so we can catch undefined, null, empty, etc. Even if we decided to change this design decision, it would be a breaking change to switch to ===, so this is why we preserve the existing status quo. At some point in the future, we might release v3.0 and that would be our opportunity to review a lot of these decisions.
